hibernate-core/src/main/java/org/hibernate/query/internal/ParameterMetadataImpl.java

```diff
@@ -36,16 +36,25 @@
 public class ParameterMetadataImpl implements ParameterMetadata {
 	private final Map<Integer,OrdinalParameterDescriptor> ordinalDescriptorMap;
 	private final Map<String,NamedParameterDescriptor> namedDescriptorMap;
+	private final Set<OrdinalParameterDescriptor> ordinalDescriptorValueCache;
+	private final Set<NamedParameterDescriptor> namedDescriptorValueCache;
 
 	public ParameterMetadataImpl(
 			Map<Integer,OrdinalParameterDescriptor> ordinalDescriptorMap,
 			Map<String, NamedParameterDescriptor> namedDescriptorMap) {
 		this.ordinalDescriptorMap = ordinalDescriptorMap == null
 				? Collections.emptyMap()
 				: Collections.unmodifiableMap( ordinalDescriptorMap );
+		this.ordinalDescriptorValueCache = ordinalDescriptorMap == null
```

Comment:
Better use `this.ordinalDescriptorMap` and test if it is empty.

Same remark below for the second one.

Reply:
ok, changing, Thanks!